### DISCUSSION TITLE
Problem switching to SSR mode

### DISCUSSION THREAD

**Discussion by rollonbears234** [satnaing/astro-paper#42] [General]
I'm new to Astro, but I followed these directions https://docs.astro.build/en/guides/deploy/deno/

And it seems to break the whole app. 

- "top level await" not allowed 
- routes are all gone, I can't load any of the posts because I get this 
```
@astrojs/sitemap: Skipped!
No pages found! We can only detect sitemap routes for "static" builds. Since you are using an SSR adapter, we recommend manually listing your sitemap routes using the "customPages" integration option.
```

What am I doing wrong? Why does everything break when the instructions in astro don't mention anything like using customPages 

I tried this on all the pages, but it didn't work either  
```
export const prerender = true
```

**Comment by satnaing** [comment]
I'm not very familiar with Deno. But after browsing some docs and GitHub issues, here's what I found.
- to integrate the project with Deno, we have to [enable SSR option](https://docs.astro.build/en/guides/server-side-rendering/#enabling-ssr-in-your-project)
- then add Deno according to [the docs](https://docs.astro.build/en/guides/integrations-guide/deno/#installation)
- [dynamic Social Image generation feature](https://astro-paper.pages.dev/posts/dynamic-og-image-generation-in-astropaper-blog-posts/) seems to have some issues with Deno. Thus, for now, we can disable this feature by removing `src/pages/[ogTitle].svg.ts` file.
- as the final step, we can define `customPages` in sitemap config like below. ([related issue](https://github.com/withastro/astro/issues/3682))
```js
export default defineConfig({
  output: "server",
  site: "https://your-site-url.com/",
  integrations: [
    tailwind({
      config: {
        applyBaseStyles: false,
      },
    }),
    react(),
    sitemap({
      customPages: ["https://your-site-url.com/posts/astro-paper-2"],
    }),
  ],
  // ---
});
```
